src/sentry_plugins/github/client.py

```diff
@@ -112,26 +147,79 @@ def delete_hook(self, repo, id):
             ),
         )
 
-    def get_last_commits(self, repo, end_sha):
-        # return api request that fetches last ~30 commits
-        # see https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository
-        # using end_sha as parameter
+    def get_installations(self):
+        # TODO(jess): remove this whenever it's out of preview
+        headers = {
+            'Accept': 'application/vnd.github.machine-man-preview+json',
+        }
+
+        params = {
+            'access_token': self.token,
+        }
+
+        return self._request('GET', '/user/installations', headers=headers, params=params)
+
+
+class GitHubIntegrationClient(GitHubClientBase):
+    url = 'https://api.github.com'
+
+    def __init__(self, installation):
+        self.installation = installation
+        self.token = None
+        self.expires_at = None
+
+    def get_token(self):
+        if not self.token or self.expires_at < datetime.datetime.utcnow():
+            res = self.create_token()
+            self.token = res['token']
+            self.expires_at = datetime.datetime.strptime(
+                res['expires_at'],
+                '%Y-%m-%dT%H:%M:%SZ',
+            )
+
+        return self.token
+
+    def get_jwt(self):
+        private_key = settings.GITHUB_INSTALLATION_PRIVATE_KEY
+
+        exp = datetime.datetime.utcnow() + datetime.timedelta(minutes=10)
+        exp = calendar.timegm(exp.timetuple())
+        # Generate the JWT
+        payload = {
+            # issued at time
+            'iat': int(time.time()),
+            # JWT expiration time (10 minute maximum)
+            'exp': exp,
+            # Integration's GitHub identifier
+            'iss': settings.GITHUB_INTEGRATION_APP_ID,
```

Comment:
Same applies here.

We could/should alternatively move these to `options` instead of settings.